apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -2201,7 +2313,27 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
     :ok = TriggersHandler.register_target(trigger_target)
 
     next_device_triggers = Map.put(device_triggers, event_type, new_targets)
+
     Map.put(state, :device_triggers, next_device_triggers)
+    |> maybe_cache_trigger_policy(trigger_target)
+  end
+
+  # TODO: consider what we should to with the cached policy if/when we allow updating a policy
+  defp maybe_cache_trigger_policy(%State{realm: realm_name} = state, %AMQPTriggerTarget{
+         parent_trigger_id: parent_trigger_id
+       }) do
```

Comment:
Don't extract in the function head if it's not necessary (cfr. [this](https://github.com/lucasvegi/Elixir-Code-Smells#complex-extractions-in-clauses))